packages/mdl-radio/mdl-radio.scss

```diff
@@ -14,120 +14,190 @@
-    cursor: pointer;
+    position: absolute;
+    // top: calc(100% - #{$mdl-radio-ui-pct} / 2);
+    // left: calc(100% - #{$mdl-radio-ui-pct} / 2);
+    width: $mdl-radio-ui-pct;
+    height: $mdl-radio-ui-pct;
     box-sizing: border-box;
+
+    &::before {
+      @include mdl-theme-prop(background-color, primary);
+
+      position: absolute;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+      transform: scale(0, 0);
+      transition:
+        opacity 120ms $mdl-animation-fast-out-linear-in-timing-function,
+        transform 120ms $mdl-animation-fast-out-linear-in-timing-function;
+      border-radius: 50%;
+      content: "";
+      opacity: 0;
+      pointer-events: none;
+    }
   }
-  // The outer circle for the radio, always present.
+
   &__outer-circle {
+    @include mdl-theme-prop(border-color, text-secondary-on-light);
+
+    @include mdl-theme-dark(".mdl-radio") {
+      @include mdl-theme-prop(border-color, text-secondary-on-dark);
+    }
+
     position: absolute;
     top: 0;
     left: 0;
-    width: $mdl-radio-size;
-    height: $mdl-radio-size;
-    transition: border-color $mdl-radio-timing-fn;
-    border: 2px solid $mdl-radio-foreground-color;
+    width: 100%;
+    height: 100%;
+    transition: border-color 120ms $mdl-animation-fast-out-linear-in-timing-function;
+    border-width: 2px;
+    border-style: solid;
     border-radius: 50%;
     box-sizing: border-box;
-    will-change: border-color;
   }
-  // The inner circle for the radio, shown when checked.
+
   &__inner-circle {
+    @include mdl-theme-prop(background-color, text-secondary-on-light);
+
+    @include mdl-theme-dark(".mdl-radio") {
+      @include mdl-theme-prop(background-color, text-secondary-on-dark);
+    }
+
     position: absolute;
     top: 0;
     left: 0;
-    width: $mdl-radio-size;
-    height: $mdl-radio-size;
-    transform: scale(0);
-    transition: transform $mdl-radio-timing-fn, background-color $mdl-radio-timing-fn;
+    width: 100%;
+    height: 100%;
+    transform: scale(0, 0);
+    transition:
+      transform 120ms $mdl-animation-fast-out-linear-in-timing-function,
+      background-color 120ms $mdl-animation-fast-out-linear-in-timing-function;
     border-radius: 50%;
-    background-color: $mdl-radio-accent-color;
     box-sizing: border-box;
-    will-change: transform;
   }
-  // Underlying native input element.
-  // Visually hidden but still able to respond to focus.
-  &__input {
+
+  &__native-control {
     position: absolute;
-    width: 1px;
-    height: 1px;
-    margin: -1px;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    margin: 0;
     padding: 0;
-    border: 0;
-    text-transform: none;
-    overflow: hidden;
-    clip: rect(0 0 0 0);
+    cursor: inherit;
+    opacity: 0;
+    z-index: 1;
   }
 }
 
-.mdl-radio__input:checked {
-  ~ .mdl-radio__container {
+.mdl-radio__native-control:checked,
+.mdl-radio__native-contro:disabled {
+  ~ .mdl-radio__background {
```

Comment:
When we tried checkbox we didn't have any problems. That being said, happy to switch to `+`. Our DOM structure dictates it, as you said, and we can get away with doing so since it's not as complicated as checkboxes.